src/glyph.rs

```diff
@@ -87,7 +87,9 @@ impl<R, F> CharacterCache for GlyphCache<R, F> where
                 }
             }
             Entry::Vacant(v) => {
-                let glyph = self.font.glyph(ch).unwrap();
+                // fallback to glyph zero or U+FFFD if glyph is not present
+                let glyph = self.font.glyph(ch).unwrap_or(self.font.glyph(rt::Codepoint(0)).unwrap_or(self.font.glyph('\u{FFFD}').unwrap()));
```

Comment:
This isn't quite right. As mentioned in the documentation for `glyph`, it will handle returning glyph zero (which would be `rt::GlyphID(0)`, not `rt::Codepoint(0)`) for you, as defined by the TrueType standard. The thing you need to check is whether, in the case that glyph zero is returned, that glyph has no shape, in which case you should try `rt::Codepoint('\u{FFFD}')`. You can check the returned glyph ID with the `id` method. To check whether there is a shape for the glyph, unfortunately I don't have a direct method for that, but you can query the shape itself or the bounding box on a `ScaledGlyph`, which will both return `None` if there is no shape.